app/commands/donations/stripe/create_payment.rb

```diff
@@ -0,0 +1,40 @@
+# This responds to a new payment that has been made
+# and creates a record of it in our database. The actual
+# creation of the payment within Stripe happens through
+# "payment intents".
+class Donations::Stripe::CreatePayment
+  class SubscriptionNotCreatedError < RuntimeError
+    def initialize
+      super "Subscription not yet created. Wait for webhook then try again."
+    end
+  end
+
+  include Mandate
+
+  initialize_with :user, :stripe_data, subscription: nil
+
+  def call
+    Donations::Payment::Create.(user, :stripe, external_id, amount_in_cents, external_receipt_url, subscription:)
+  end
+
+  private
+  def external_id = stripe_data.id
+  def external_receipt_url = stripe_data.charges.first.receipt_url
+  def amount_in_cents = stripe_data.amount
+
+  memoize
+  def subscription
+    return @subscription if @subscription
+
+    return unless stripe_data.invoice
```

Comment:
```suggestion
    return nil unless stripe_data.invoice
```

I just prefer being explicit here as the return type actually matters.